### PR TITLE
Bug fix: fix Iterable "update in place" methods of BuiltList and BuiltSet so they discard original list or set.

### PR DESCRIPTION
**Pull request by davidmorgan**

### CODE REVIEW

File: test/list/list_builder_test.dart [hunk 1]
```diff
@@ -200,41 +218,57 @@ void main() {
...
     });
 
     test('has a method like List.removeAt that returns nothing', () {
       expect((new ListBuilder<int>([1, 2])..removeAt(1)).build(), [1]);
-    });
-
-    test('has a method like List.removeAt that returns nothing', () {
-      expect((new ListBuilder<int>([1, 2])..removeAt(1)).build(), [1]);
+      expect(
+          (new BuiltList<int>([1, 2]).toBuilder()..removeAt(1)).build(), [1]);
```
Comment (BlackHC) [inline test/list/list_builder_test.dart]: Do we want to check that it returns nothing?

File: lib/src/list_multimap/list_multimap_builder.dart [hunk 2]
```diff
@@ -125,30 +125,38 @@ class ListMultimapBuilder<K, V> {
...
   }
 
   // Internal.
 
-  ListMultimapBuilder._fromBuiltListMultimap(
-      BuiltListMultimap<K, V> builtListMultimap)
-      : _copyBeforeWrite = true,
-        _builtListMultimap = builtListMultimap,
-        _builtMap = builtListMultimap._map {
+  void _disown() {
```
Comment (dskloetg) [inline lib/src/list_multimap/list_multimap_builder.dart]: This is not a clear name. makeWritableCopy?

File: lib/src/list/list_builder.dart [hunk 3]
```diff
@@ -60,176 +58,150 @@ class ListBuilder<E> {
...
-    _list.replaceRange(start, end, iterable);
+    _safeList.replaceRange(start, end, iterable);
   }
 
   // Based on Iterable.
 
   /// As [Iterable.map], but updates the builder in place. Returns nothing.
   void map(E f(E element)) {
-    _list = _list.map(f).toList(growable: true);
+    _setSafeList(_list.map(f).toList(growable: true));
```
Comment (BlackHC) [inline lib/src/list/list_builder.dart]: set  _safeList ...?

File: lib/src/list_multimap/list_multimap_builder.dart [hunk 4]
```diff
@@ -12,10 +12,9 @@ part of built_collection.multimap;
 /// [Built Collection library documentation](#built_collection/built_collection)
 /// for the general properties of Built Collections.
 class ListMultimapBuilder<K, V> {
-  bool _copyBeforeWrite;
-  BuiltListMultimap<K, V> _builtListMultimap;
   Map<K, BuiltList<V>> _builtMap;
-  Map<K, ListBuilder<V>> _builderMap = new Map<K, ListBuilder<V>>();
+  BuiltListMultimap<K, V> _builtMapOwner;
```
Comment (BlackHC) [inline lib/src/list_multimap/list_multimap_builder.dart]: I prefer _builtListMultimap over _builtMapOwner (same for the other Builders).

I think while correct the *Owner bit is a foreign concept that makes it harder to understand the code here.

Why not: setList, setBuiltList instead setSafeList, setListOwner?

Comment (davidmorgan) [inline lib/src/list_multimap/list_multimap_builder.dart]: I like this better, but maybe there's a better option still. We can discuss at length later :)

File: test/list/list_builder_test.dart (hunk 1 above)
Comment (davidmorgan) [inline test/list/list_builder_test.dart]: Filed an issue.

**Pull request merged**